templates/repo/diff/compare.tmpl

```diff
@@ -12,11 +12,32 @@
 				{{.i18n.Tr "action.compare_commits_general"}}
 			{{ end }}
 		</h2>
+		{{ $BaseCompareName := $.BaseName -}}
+		{{- $HeadCompareName := $.HeadRepo.OwnerName -}}
+		{{- if and (eq $.BaseName $.HeadRepo.OwnerName) (ne $.Repository.Name $.HeadRepo.Name) -}}
+			{{- $HeadCompareName = printf "%s/%s" $.HeadRepo.OwnerName $.HeadRepo.Name -}}
+		{{- end -}}
+		{{- $OwnForkCompareName := "" -}}
+		{{- if .OwnForkRepo -}}
+			{{- $OwnForkCompareName = .OwnForkRepo.OwnerName -}}
+		{{- end -}}
+		{{- $RootRepoCompareName := "" -}}
+		{{- if .RootRepo -}}
+			{{- $RootRepoCompareName = .RootRepo.OwnerName -}}
+			{{- if eq $.HeadRepo.OwnerName .RootRepo.OwnerName -}}
+				{{- $HeadCompareName = printf "%s/%s" $.HeadRepo.OwnerName $.HeadRepo.Name -}}
+			{{- end -}}
+			{{- if .OwnForkRepo -}}
+				{{- if eq $.OwnForkRepo.OwnerName .RootRepo.OwnerName -}}
+					{{- $OwnForkRepoCompareName = printf "%s/%s" $.OwnForkRepo.OwnerName $.OwnForkRepo.Name -}}
+				{{- end -}}
+			{{- end -}}
+		{{- end -}}
```

Comment:
why not define them in the golang file?

Reply:
Have you seen compare.go?

Reply:
However more seriously:

This is purely UI. It has no effect on the code, or even URLs that are rendered.

Any future JavaScript that is used to render branches asynchronously will need to do this work.

Placing this in the go code will make the go code even more complex and less transparent.

I don't believe that it would be any more easy to understand what is going on being detached from the rendering in that way.